misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/utils/HttpClientRequest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.com). All Rights Reserved.
+ *
+ * This software is the property of WSO2 Inc. and its suppliers, if any.
+ * Dissemination of any information or reproduction of any material contained
+ * herein is strictly forbidden, unless permitted by WSO2 in accordance with
+ * the WSO2 Commercial License available at http://wso2.com/licenses.
+ * For specific language governing the permissions and limitations under
+ * this license, please see the license as well as any agreement you’ve
+ * entered into with WSO2 governing the purchase of this software and any
+ */
+package org.ballerinalang.datamapper.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * This class can be used to send http request.
+ */
+public class HttpClientRequest {
+    private static final Logger LOG = LoggerFactory.getLogger(HttpClientRequest.class);
+    private static final int DEFAULT_READ_TIMEOUT = 30000;
+
+    /**
+     * Send an HTTP POST request to a service.
+     *
+     * @param endpoint - service endpoint
+     * @param postBody - message payload
+     * @param headers  http request headers map
+     * @return - HttpResponse from end point
+     * @throws IOException If an error occurs while sending the GET request
+     */
+    public static HttpResponse doPost(String endpoint, String postBody, Map<String, String> headers)
+            throws IOException {
+        HttpURLConnection urlConnection = null;
+        try {
+            urlConnection = getURLConnection(endpoint);
+            setHeadersAndMethod(urlConnection, headers, "POST");
+            try (OutputStream out = urlConnection.getOutputStream();
+                 Writer writer = new OutputStreamWriter(out, StandardCharsets.UTF_8)) {
+                writer.write(postBody);
+            }
+            return buildResponse(urlConnection);
+        } finally {
+            if (urlConnection != null) {
+                urlConnection.disconnect();
+            }
+        }
+    }
+
+    private static HttpURLConnection getURLConnection(String requestUrl) throws IOException {
+        return getURLConnection(requestUrl, DEFAULT_READ_TIMEOUT);
+    }
+
+    private static HttpURLConnection getURLConnection(String requestUrl, int readTimeout) throws IOException {
```

Comment:
Why `readTimeout` is passed as a parameter as it is constant. Also for connect time out also, we have used a constant value(15000) but it is not passed as a parameter or defined as a constant. Can we follow the same convention for all the properties?

Reply:
Made the changes to follow the same convention.